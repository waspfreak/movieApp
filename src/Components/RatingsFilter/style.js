import styled from 'styled-components';

export const StyledRatingsFilter = styled.div`
	width: 30%;
	position: absolute;
	left: 30%;
	padding-left: 20px;

	.star-wrapper {
		display: flex;
		place-content: left;
		filter: drop-shadow(0 3px 3px rgba(0, 0, 0, 0.5));
	}

	.label {
		color: white;
		font-size: 16px;
		padding-left: 8px;
	}

	.star-group {
		font-size: 0;
		direction: rtl;
		margin-top: 0.55rem;
	}

	.star-group * {
		font-size: 2rem;
	}

	.star-group > input {
		display: none;
	}

	.star-group > input + label {
		display: inline-block;
		overflow: hidden;
		text-indent: 9999px;
		width: 1.65rem;
		white-space: nowrap;
		cursor: pointer;
		font-size: 1.5rem;
	}

	.star-group > input + label:before {
		display: inline-block;
		text-indent: -9999px;
		content: '☆';
		color: #111;
		transition: all ease 0.2s;
	}

	.star-group > input:checked ~ label:before,
	.star-group > input + label:hover ~ label:before,
	.star-group > input + label:hover:before {
		content: '★';
		color: rgba(218, 50, 28, 0.9);
		text-shadow: 0 0 1px rgba(32, 32, 32, 0.5);
	}

	.star-group > .star-empty + label {
		text-indent: -9999px;
		width: 0.25em;
	}

	.star-group > .star-empty + label:before {
		width: 0.25em;
	}

	.star-group:hover > input + label:before {
		content: '☆';
		color: #111;
		text-shadow: none;
	}

	.star-group:hover > input + label:hover ~ label:before,
	.star-group:hover > input + label:hover:before {
		content: '★';
		color: rgba(218, 50, 28, 0.9);
		text-shadow: 0 0 1px rgba(32, 32, 32, 0.5);
	}

	@media screen and (max-width: 748px) {
		width: 100%;
		position: relative;
		left: initial;
		padding-left: 0px;
		.label {
			padding-left: 0;
		}
	}
`;
