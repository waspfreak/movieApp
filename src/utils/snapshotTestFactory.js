import {cleanup, render} from '@testing-library/react';
import React from 'react';

/**
 *
 * Sets up snapshot tests in a standard way and avoids repeating boiler plate
 */
export function snapshotTestFactory(Component: any, properties: any) {
	/**
	 * Re-usable it block
	 */
	const itBlock = (props: any) => {
		it('should match snapshot', () => {
			const {container} = render(<Component {...props} />);
			expect(container).toMatchSnapshot();
		});
	};

	/**
	 * Re-usable describe block
	 */
	describe(`${Component.name}`, () => {
		/**
		 * Clean up any mock functions that might be in props
		 */
		afterEach(() => {
			jest.clearAllMocks();
			cleanup();
		});

		/**
		 * If its an array of props then iterate them into seperate it blocks
		 */
		if (Array.isArray(properties)) {
			properties.forEach((props) => {
				itBlock(props);
			});
		} else {
			itBlock(properties);
		}
	});
}
