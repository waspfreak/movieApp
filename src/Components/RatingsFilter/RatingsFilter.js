import React, {useEffect, useState} from 'react';

// Components
import MovieCard from '../Card/MovieCard';
import MovieWall from '../MovieWall/MovieWall';

// Environment Variables
import {FILTER_BASE_URL, IMAGE_BASE_URL, POSTER_SIZE} from '../../api/API';

// Styled Components
import {StyledRatingsFilter} from './style';

/**
 * Ratings Filter UI component for user interaction
 */

export const RatingsFilter = () => {
	const [selected, setSelected] = useState(false);
	const [ratings, setRatings] = useState([]);

	useEffect(() => {
		async function fetchData() {
			const response = await fetch(FILTER_BASE_URL);
			const data = await response.json();
			setRatings(data.results);
		}
		fetchData();
	}, []);

	const handleInputChange = (e) => {
		setSelected(e.target.id);
	};

	const filteredFiveStars = (
		<MovieWall children>
			{ratings
				.filter((movie) => movie.vote_average > 8)
				.map((movie) => (
					<>
						<div key={movie.id}>
							<MovieCard
								movieId={movie.id}
								clickable
								image={
									movie.poster_path
										? IMAGE_BASE_URL + POSTER_SIZE + movie.poster_path
										: null
								}
								originalTitle={movie.original_title}
								voteAverage={parseFloat(movie.vote_average).toFixed(1) * 10}
							/>
						</div>
					</>
				))}
		</MovieWall>
	);

	const filteredFourStars = (
		<div className={selected ? 'BOX' : ''}>
			<MovieWall children>
				{ratings
					.filter((movie) => movie.vote_average <= 8 && movie.vote_average >= 6)
					.map((movie) => (
						<>
							<div key={movie.id}>
								<MovieCard
									key={movie.id}
									clickable
									image={
										movie.poster_path
											? IMAGE_BASE_URL + POSTER_SIZE + movie.poster_path
											: null
									}
									movieId={movie.id}
									originalTitle={movie.original_title}
									voteAverage={parseFloat(movie.vote_average).toFixed(1) * 10}
								/>
							</div>
						</>
					))}
			</MovieWall>
		</div>
	);

	const filteredThreeStars = (
		<div className={selected ? 'BOX' : ''}>
			<MovieWall children>
				{ratings
					.filter((movie) => movie.vote_average <= 6 && movie.vote_average >= 4)
					.map((movie) => (
						<>
							<div key={movie.id}>
								<MovieCard
									key={movie.id}
									clickable
									image={
										movie.poster_path
											? IMAGE_BASE_URL + POSTER_SIZE + movie.poster_path
											: null
									}
									movieId={movie.id}
									originalTitle={movie.original_title}
									voteAverage={parseFloat(movie.vote_average).toFixed(1) * 10}
								/>
							</div>
						</>
					))}
			</MovieWall>
		</div>
	);

	const filteredTwoStars = (
		<MovieWall children>
			{ratings
				.filter((movie) => movie.vote_average <= 4 && movie.vote_average >= 2)
				.map((movie) => (
					<>
						<div key={movie.id}>
							<MovieCard
								key={movie.id}
								clickable
								image={
									movie.poster_path
										? IMAGE_BASE_URL + POSTER_SIZE + movie.poster_path
										: null
								}
								movieId={movie.id}
								originalTitle={movie.original_title}
								voteAverage={parseFloat(movie.vote_average).toFixed(1) * 10}
							/>
						</div>
					</>
				))}
		</MovieWall>
	);

	const filteredOneStar = (
		<MovieWall children>
			{ratings
				.filter((movie) => movie.vote_average <= 2 && movie.vote_average >= 0)
				.map((movie) => (
					<>
						<div key={movie.id}>
							<MovieCard
								clickable
								key={movie.id}
								image={
									movie.poster_path
										? IMAGE_BASE_URL + POSTER_SIZE + movie.poster_path
										: null
								}
								movieId={movie.id}
								originalTitle={movie.original_title}
								voteAverage={parseFloat(movie.vote_average).toFixed(1) * 10}
							/>
						</div>
					</>
				))}
		</MovieWall>
	);

	return (
		<>
			<StyledRatingsFilter>
				<label className='label'>Filter movie ratings</label>
				<section className='star-wrapper'>
					<span className='star-group'>
						<input
							onClick={handleInputChange}
							type='radio'
							id='rating-5'
							name='rating'
						/>
						<label htmlFor='rating-5'>5</label>

						<input
							onClick={handleInputChange}
							type='radio'
							id='rating-4'
							name='rating'
						/>
						<label htmlFor='rating-4'>4</label>

						<input
							onClick={handleInputChange}
							type='radio'
							id='rating-3'
							name='rating'
						/>
						<label htmlFor='rating-3'>3</label>

						<input
							onClick={handleInputChange}
							type='radio'
							id='rating-2'
							name='rating'
						/>
						<label htmlFor='rating-2'>2</label>

						<input
							onClick={handleInputChange}
							type='radio'
							id='rating-1'
							name='rating'
						/>
						<label htmlFor='rating-1'>1</label>

						<input
							type='radio'
							id='default'
							name='rating'
							className='star-empty'
						/>
						<label htmlFor='default-state'>empty</label>
					</span>
				</section>
			</StyledRatingsFilter>
			<div>
				{selected === 'rating-5'
					? filteredFiveStars
					: selected === 'rating-4'
					? filteredFourStars
					: selected === 'rating-3'
					? filteredThreeStars
					: selected === 'rating-2'
					? filteredTwoStars
					: selected === 'rating-1'
					? filteredOneStar
					: null}
			</div>
		</>
	);
};

export default RatingsFilter;
