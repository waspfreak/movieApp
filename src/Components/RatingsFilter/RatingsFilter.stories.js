import React from 'react';

import {RatingsFilter} from './RatingsFilter';

export default {
	title: 'Example/RatingsFilter',
	component: RatingsFilter,
};

const Template = (args) => <RatingsFilter {...args} />;

export const RatingsFilterExample = Template.bind({});
RatingsFilterExample.args = {};
