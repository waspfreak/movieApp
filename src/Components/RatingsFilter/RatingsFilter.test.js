import {snapshotTestFactory} from '../../utils/snapshotTestFactory';

import {RatingsFilter} from './RatingsFilter';
const props: RatingsFilter[] = [{}];
snapshotTestFactory(RatingsFilter, props);
