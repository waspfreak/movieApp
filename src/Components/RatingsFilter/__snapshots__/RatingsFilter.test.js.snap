// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsFilter should match snapshot 1`] = `
.c0 {
  font-family: 'Titillium Web',sans-serif;
  width: 30%;
  position: absolute;
  left: 30%;
  padding-left: 20px;
}

.c0 .star-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  place-content: left;
  -webkit-filter: drop-shadow(0 3px 3px rgba(0,0,0,0.5));
  filter: drop-shadow(0 3px 3px rgba(0,0,0,0.5));
}

.c0 .label {
  color: white;
  font-size: 16px;
  padding-left: 8px;
}

.c0 .star-group {
  font-size: 0;
  direction: rtl;
  margin-top: 0.55rem;
}

.c0 .star-group * {
  font-size: 2rem;
}

.c0 .star-group > input {
  display: none;
}

.c0 .star-group > input + label {
  display: inline-block;
  overflow: hidden;
  text-indent: 9999px;
  width: 1.65rem;
  white-space: nowrap;
  cursor: pointer;
  font-size: 1.5rem;
}

.c0 .star-group > input + label:before {
  display: inline-block;
  text-indent: -9999px;
  content: '☆';
  color: #111;
  -webkit-transition: all ease 0.2s;
  transition: all ease 0.2s;
}

.c0 .star-group > input:checked ~ label:before,
.c0 .star-group > input + label:hover ~ label:before,
.c0 .star-group > input + label:hover:before {
  content: '★';
  color: rgba(218,50,28,0.9);
  text-shadow: 0 0 1px rgba(32,32,32,0.5);
}

.c0 .star-group > .star-empty + label {
  text-indent: -9999px;
  width: 0.25em;
}

.c0 .star-group > .star-empty + label:before {
  width: 0.25em;
}

.c0 .star-group:hover > input + label:before {
  content: '☆';
  color: #111;
  text-shadow: none;
}

.c0 .star-group:hover > input + label:hover ~ label:before,
.c0 .star-group:hover > input + label:hover:before {
  content: '★';
  color: rgba(218,50,28,0.9);
  text-shadow: 0 0 1px rgba(32,32,32,0.5);
}

@media screen and (max-width:748px) {
  .c0 {
    width: 100%;
    position: relative;
    left: initial;
    padding-left: 0px;
  }

  .c0 .label {
    padding-left: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <label
      class="label"
    >
      Filter movie ratings
    </label>
    <section
      class="star-wrapper"
    >
      <span
        class="star-group"
      >
        <input
          id="rating-5"
          name="rating"
          type="radio"
        />
        <label
          for="rating-5"
        >
          5
        </label>
        <input
          id="rating-4"
          name="rating"
          type="radio"
        />
        <label
          for="rating-4"
        >
          4
        </label>
        <input
          id="rating-3"
          name="rating"
          type="radio"
        />
        <label
          for="rating-3"
        >
          3
        </label>
        <input
          id="rating-2"
          name="rating"
          type="radio"
        />
        <label
          for="rating-2"
        >
          2
        </label>
        <input
          id="rating-1"
          name="rating"
          type="radio"
        />
        <label
          for="rating-1"
        >
          1
        </label>
        <input
          class="star-empty"
          id="default"
          name="rating"
          type="radio"
        />
        <label
          for="default-state"
        >
          empty
        </label>
      </span>
    </section>
  </div>
  <div />
</div>
`;
